lib/src/components/NxSearchDropdown/__tests__/NxSearchDropdown.test.tsx

```diff
@@ -0,0 +1,321 @@
+            { attachTo: mountPoint }
+        );
+
+        (component.find('button.nx-load-error__retry').getDOMNode() as HTMLElement).focus();
+
+        component.setProps({ error: null });
+
+        expect(component.find('.nx-load-error__retry')).not.toExist();
+        expect(document.activeElement === component.find('.nx-search-dropdown__input input').getDOMNode()).toBe(true);
+      }
+  );
+
+  it('calls onSearch with the current trimmed searchText if focus enters the component from elsewhere on the page ' +
+      'while there is an error', function() {
+    const onSearch = jest.fn(),
+        component = getMounted(
+            { searchText: 'foo ', error: 'bar', onSearch },
+            { attachTo: mountPoint }
+        );
+
+    const anotherElement = document.createElement('button');
+    document.body.append(anotherElement);
+    anotherElement.focus();
+
+    expect(onSearch).not.toHaveBeenCalled();
+
+    (component.find('.nx-search-dropdown__input input').getDOMNode() as HTMLElement).focus();
+
+    expect(onSearch).toHaveBeenCalledWith('foo');
+
+    anotherElement.remove();
+  });
+
+  it('does not call onSearch if focus moves within the component while there is an error', function() {
+    const onSearch = jest.fn(),
+        component = getMounted(
+            { searchText: 'foo', onSearch },
+            { attachTo: mountPoint }
+        ),
+        filterInput = component.find('.nx-search-dropdown__input input').getDOMNode() as HTMLElement;
+
+    expect(onSearch).not.toHaveBeenCalled();
+
+    // get focus into the component before we set the error
+    filterInput.focus();
+    expect(onSearch).not.toHaveBeenCalled();
+
+    component.setProps({ error: 'err' });
+    expect(onSearch).not.toHaveBeenCalled();
+
+    const retryButton = component.find('button.nx-load-error__retry').getDOMNode() as HTMLElement;
+    retryButton.focus();
+    expect(onSearch).not.toHaveBeenCalled();
+
+    filterInput.focus();
+    expect(onSearch).not.toHaveBeenCalled();
+  });
+
+  it('does not call doSeach if focus moves into the component from an outside window while there is an error',
+      function() {
+        const onSearch = jest.fn(),
+            component = getMounted(
+                { searchText: 'foo', onSearch },
+                { attachTo: mountPoint }
+            ),
+            filterInput = component.find('.nx-search-dropdown__input input').getDOMNode() as HTMLElement;
+
+        expect(onSearch).not.toHaveBeenCalled();
+
+        // get focus into the component before we set the error
+        filterInput.focus();
+        expect(onSearch).not.toHaveBeenCalled();
+
+        component.setProps({ error: 'err' });
+        expect(onSearch).not.toHaveBeenCalled();
+
+        (document.activeElement! as HTMLElement).blur();
+        expect(onSearch).not.toHaveBeenCalled();
+
+        filterInput.focus();
+        expect(onSearch).not.toHaveBeenCalled();
+      }
+  );
+});
```

Comment:
Should we add the test where if there's an error querying the data, and if the user navigates away from the component and comes back to it, the search should be retried automatically (as mentioned [here](https://github.com/sonatype/sonatype-react-shared-components/pull/467/files#diff-22194081f00f47900c6cba95001a93b3add1e569d6189026fe59d27a50f7ffc9R65-R70)). 

There is a test for calling `onSearch` with the current trimmed searchText if focus enters the component from elsewhere, but we don't have an explicit case when there's an error in querying and focus enters from elsewhere.

Reply:
> Should we add the test where if there's an error querying the data, and if the user navigates away from the component and comes back to it, the search should be retried automatically (as mentioned here).
>
> There is a test for calling onSearch with the current trimmed searchText if focus enters the component from elsewhere, but we don't have an explicit case when there's an error in querying and focus enters from elsewhere.

Is that not what https://github.com/sonatype/sonatype-react-shared-components/pull/467/files#diff-8faf831d9f7a4427b8d4674137223bd7eb914e2b245eddc1308562fa0ce5a3f9R250 is?  That behavior is only supposed to happen when there's an error so if you're saying we have a test that asserts that it happens when there _isn't_ an error, that'd be a problem.